1. Reverse a given number loops and equations and return true if it is the same as the original
number.

In [ ]:
while True:
    try:
        x=int(input('Enter an integer number: '))
        if x or x==0:
            break
    except:
        print('only Numbers Are valid!!')

m=float(x)
y=0

while True:
    if m<0:
        print(y)
        break
    y=(y*10)+ (m %10)
    m=m/10
    

Enter an integer number: 98


2. Return the number of times that the string “Emma” appears anywhere in the given string.

In [7]:
sr=input('Enter a string: ')
print(sr.count('Emma'))

Enter a string: Emma mahmoud 446 Emma Emma
3


3. Print the following pattern.
12
2
3 3 3
4 4 4 4
5 5 5 5 5

In [12]:
for n in range(1,6,1):
        print(str(n)*n )

1
22
333
4444
55555


4. Given a string of odd length greater than 7, return a string made of the middle three chars of
a given String.

In [21]:
sr='qwertyuioplkjhgfdasb'
x= int(len(sr)/2)
b=x-1
d=x+2
c=sr[b:d]
print(c)

plk


5. Given 2 strings, s1 and s2, create a new string by appending s2 in the middle of s1.

In [30]:
s1='mahmoud'
s2=' alqannas '
x=int(len(s1)/2)
s3=s1[:x]+s2+s1[x:]
print(s3)


mah alqannas moud


6. Given 2 strings, s1, and s2 return a new string made of the first, middle and last char each
input string.

In [44]:
s1='12345'
s2='salem'
x1=int((len(s1)/2))
x2=int((len(s2)/2))
s3=s1[0]+s1[x1]+s1[len(s1)-1]+s2[0]+s2[x2]+s2[len(s2)-1]
print(s3)

135slm


7. arrange String characters such that lowercase letters should come first.

In [45]:
sr='MAfklksdAA456as'
print(sorted(sr,key = str.lower))

['4', '5', '6', 'A', 'A', 'A', 'a', 'd', 'f', 'k', 'k', 'l', 'M', 's', 's']


8. Sort a given string alphabetically.

In [47]:
st='mahmoud kkksss'
print(sorted(st))

[' ', 'a', 'd', 'h', 'k', 'k', 'k', 'm', 'm', 'o', 's', 's', 's', 'u']


9. Given a string input Count all lower case, upper case, digits, and special symbols
Sample Input: ”My?Name!Is956Yazeed”
Sample Output:
Number of upper case: 4
Number of lower case: 10
Number of digits: 3
Number of special symbols: 2

In [57]:
st='My?Name!Is956Yazeed'
upper1=0
lower1=0
num=0
special=0
for n in st:
    b=str(n)
  
    if(b.isupper()):
       upper1=upper1 +1
    elif(b.islower()):
        lower1=lower1+1
    elif(b.isalnum()):
        num=num+1
    else:
        special=special+1
        
print(' Number of upper case: ',upper1)
print(' Number of lower case: ',lower1)    
print(' Number of digit: ',num)
print(' Number of special symbols: ',special)

 Number of upper case:  4
 Number of lower case:  10
 Number of digit:  3
 Number of special symbols:  2


10. Write python program that takes a positive integer n as an input from the user and calculates the
sum up to n.

In [1]:
while True:
    try:
        x=int(input('Enter an integer number: '))
        if x or x==0:
            break
    except:
        print('only Numbers Are valid!!')
x1=x+1
sum1=0
for m in range(x1):
    sum1=sum1+m
print(sum1)
    
    

        
        
        
   
    

Enter an integer number: hjg
only Numbers Are valid!!
Enter an integer number: 5
15


11. Use the While or for statement to write a program that computes and displays the factorial of
a positive integer n entered by the user.

In [3]:
while True:
    try:
        x=int(input('Enter an integer number: '))
        if x<0:
            break
        if x or x==0:
            break
    except:
        print('only positive Numbers Are valid!!')
num=1
for n in  range(x):
    num=num*(n+1)
print(num)
    



Enter an integer number: 5
120


12. Write a program that uses a while loop to display values from -5 through 5, using an increment
of 0.5.

In [3]:
n=-5.5

while n<5:
    n=n+.5
    print(n)
    
    

-5.0
-4.5
-4.0
-3.5
-3.0
-2.5
-2.0
-1.5
-1.0
-0.5
0.0
0.5
1.0
1.5
2.0
2.5
3.0
3.5
4.0
4.5
5.0


13. Write a c program that prints the following star pattern:
* * * *
* * *
* *
*

In [4]:
for n in range(4):
    print('* '*(4-n))

* * * * 
* * * 
* * 
* 


14. Write a c program that prints the following star pattern:
**
*
* * *
* * * *
* * * * *
* * * *
* * *
* *
*

In [1]:
x=6
v=x-2
for m in range(x):
    print('* '*m)
for n in range(v):
    print('* '*(v-n))
  
    
        
    


* 
* * 
* * * 
* * * * 
* * * * * 
* * * * 
* * * 
* * 
* 


15. Write a c program that prints the following star pattern:
7*
6
* * 5
* * * 4
* * * * 3

In [57]:
x=8
for b in range(5):
    x=x-1
    print('*'*(7-x), x)
    

 7
* 6
** 5
*** 4
**** 3


16. Write a c program that prints the following star pattern:
        

In [70]:
for n in range(6):
    print(' '*(6-n),'* '*(n))


       
      * 
     * * 
    * * * 
   * * * * 
  * * * * * 


17. Write a python program that prints the following star pattern:

In [74]:
for n in range(5):
    print(' '*(n),'* '*(5-n))

 * * * * * 
  * * * * 
   * * * 
    * * 
     * 


18. Write a python program that prints the following star pattern:

In [79]:
for n in range(5):
    print(' '*(5-n),'* '*(n))
for n in range(5):
    print(' '*(n),'* '*(5-n))

      
     * 
    * * 
   * * * 
  * * * * 
 * * * * * 
  * * * * 
   * * * 
    * * 
     * 


19. Write programs that implement the following equations:
    

In [2]:
m=100
d=94
n=0
n1=6
for s in range(m):
    n =n + (s**2)
eq1=n
for b in range(6,101):
    n1 =n1 +(b**2)*(b-5)
eq2=n1
print('result of eqation number 1 is: ', eq1)
print('result of eqation number 2 is: ', eq2)

result of eqation number 1 is:  328350
result of eqation number 2 is:  23810806


20. Write a Python program to print Fibonacci series up to 10th term.

In [11]:
while True:
    try:
        x=int(input('Enter an integer number: '))
        if x or x > 0:
            break
    except:
        print('Enter a valid  positive number: ')
        
if x ==0:
    print('Fibonacci of 1 = 1')
elif x== 1:
     print('Fibonacci of 1 = 1')
else:
    a=0
    b=1
    for n in range(2,x):
        c=a+b
        a=b
        b=c
    f=b

    print("Fibonacci of {} =".format(x) , f)
    

Enter an integer number: 9
Fibonacci of 9 = 21


21. Write a Python program to print GCD between any two given numbers.

In [15]:
import math
x= 40
v=60
print(math.gcd(x,v))

20


22. Write a Python program to print LCM between any two given numbers.

In [3]:
while True:
    try:
        x=int(input('Enter an integer x: '))
        if x or x==0:
            break
    except:
        print('only Numbers Are valid!!')
        
while True:
    try:
        y=int(input('Enter an integer number: '))
        if x or y==0:
            break
    except:
        print('only Numbers Are valid!!')
  
def lcm(x,y):
    

    if x>y:
        greater=x
    elif y<=x:
        greater=y

    while(True):
        
        if((greater % x == 0) and (greater % y ==0)):
            lcm =greater
            break
        greater+=1
    return lcm    

print('LCM between x and y is: ',lcm(x,y))

Enter an integer x: 1
Enter an integer number: 1
LCM between x and y is:  1


23. Write a Python program to find the sum of all natural odd numbers between 1 and 100.

In [6]:
sum1=0
for n in range(101):
    if n%2!=0:
        sum1=sum1+n
print(sum1)
        

2500


###### 24. Write a Python program to find the sum of all natural prime numbers between 1 and 100.

In [28]:
prime=[True]*101
p=2
while p*p <=100:
    if prime[p] == True:
        i =p*2
        #print(prime)
        while i <=n :
            prime[i]= False
            i+=p
            #print(prime)
    p+=1    
sum = 0
for s in range(2,101):
    if (prime[s]):
        sum+=s
print(sum)
        


1060


25. Write a Python program to find the product of all natural even numbers between 1 and 100.

In [29]:
sum1=0
for n in range(101):
    if n%2==0:
        sum1=sum1+n
print(sum1)
        

2550


26. Write a Python program that prints the multiplication tables of all numbers between 0 and 9.

In [32]:
n=10
for i in range(10):
    for j in range(n):
        print("{} * {} =  ".format(i,j),i*j)

0 * 0 =   0
0 * 1 =   0
0 * 2 =   0
0 * 3 =   0
0 * 4 =   0
0 * 5 =   0
0 * 6 =   0
0 * 7 =   0
0 * 8 =   0
0 * 9 =   0
1 * 0 =   0
1 * 1 =   1
1 * 2 =   2
1 * 3 =   3
1 * 4 =   4
1 * 5 =   5
1 * 6 =   6
1 * 7 =   7
1 * 8 =   8
1 * 9 =   9
2 * 0 =   0
2 * 1 =   2
2 * 2 =   4
2 * 3 =   6
2 * 4 =   8
2 * 5 =   10
2 * 6 =   12
2 * 7 =   14
2 * 8 =   16
2 * 9 =   18
3 * 0 =   0
3 * 1 =   3
3 * 2 =   6
3 * 3 =   9
3 * 4 =   12
3 * 5 =   15
3 * 6 =   18
3 * 7 =   21
3 * 8 =   24
3 * 9 =   27
4 * 0 =   0
4 * 1 =   4
4 * 2 =   8
4 * 3 =   12
4 * 4 =   16
4 * 5 =   20
4 * 6 =   24
4 * 7 =   28
4 * 8 =   32
4 * 9 =   36
5 * 0 =   0
5 * 1 =   5
5 * 2 =   10
5 * 3 =   15
5 * 4 =   20
5 * 5 =   25
5 * 6 =   30
5 * 7 =   35
5 * 8 =   40
5 * 9 =   45
6 * 0 =   0
6 * 1 =   6
6 * 2 =   12
6 * 3 =   18
6 * 4 =   24
6 * 5 =   30
6 * 6 =   36
6 * 7 =   42
6 * 8 =   48
6 * 9 =   54
7 * 0 =   0
7 * 1 =   7
7 * 2 =   14
7 * 3 =   21
7 * 4 =   28
7 * 5 =   35
7 * 6 =   42
7 * 7 =   49
7 * 8 =   56
7 * 9 =   6

27. Write a Python program to input a number from user and count number of digits in the given
integer.

In [3]:
while True:
    try:
        x=int(input('Enter a number: '))
        if x or x == 0:
            break
    except:
        print('Enter a valid number:')
b=0
n=10
while (x!=0):
    x=int(x/n)

    b+=1
    
print(b)

Enter a number: 6489
4


28. Write a Python program to input a number from user and find if the number is Palindrome.

In [7]:
while True:
    try:
        x=int(input('Enter a number: '))
        if x or x == 0:
            break
    except:
        print('Enter a valid number:')
dev=0
temp=x
while (x>0):
    dig=x%10
    dev=dev*10+dig
    x//=10
if (temp== dev):
    print('The entered number is Palindrome')
else:
    print('The entered number is NOT Palindrome!!!')

Enter a number: 54
4
45
The entered number is NOT Palindrome!!!


29. Write a Python program to input a number and calculate the sum of its digits.

In [3]:
while True:
    try:
        x=int(input('Enter a number: '))
        if x or x == 0:
            break
    except:
        print('Enter a valid number:')
dev=0
temp=x
d=0
s=0
while (x>0):
    dig=x%10
    dev=dev*10+dig
    s=s+dev%10
    d+=1
   # print(dev)
    x//=10
print(s)


Enter a number: 987
24


30. Write a Python program find the maximum number to get of 3-3 digit palindrome(do not use
strings).

31. Given a string, return the sum and average of the digits that appear in the string, ignoring all
other characters.

In [14]:
st='mahmoud8f ff45688d'
b=list(st)
m=len(st)
sum1=0
av=0
for n in range(m):
    s=str(b[n])
    if (s.isdecimal()):
        av+=1
        
        sum1=sum1+int(b[n])
    
print( 'sum =',sum1)       
print('average= ', (sum1/av))

8
4
5
6
8
8
sum = 39
average=  6.5


32. Find all occurrences of “USA” in given string ignoring the case.

In [38]:
st = 'mamfdkusa usa USA USA UHW UsA'
st=st.upper()
b=st.count('USA')
print(b)

5


33. Apply the String characters balance Test (occurrences of all appeared characters must be equal).

In [27]:
sts=input('Enter a string ')
st=sts
st1='aasddf'
s=0
d=0
b =list()
for n in st:
    b.append(int(str(st.count(n))))
b=list(dict.fromkeys(b)) 
if (len(b)==1):
        print('the String characters balance')
else:
        print('the String is NOT characters balance !!!!')
    
    


Enter a string mkh
the String characters balance


34. Given a string and and int n, remove n characters from string by each time until the string has
no more characters.

In [22]:
 n=int(input('Enter a number: '))
 st='mahmoudmma'
# c=len(st)
# while (c>0):
#     st=st-st[:n]
#     c=len(st)
#     print(st)

b=list(st)
c=len(b)+1
while (c>0):
        st=str(b)
        c=len(b)
        st=st.replace(',','')
        print(st)
        del b[:n]
        
        
        



Enter a number: 2
['m' 'a' 'h' 'm' 'o' 'u' 'd' 'm' 'm' 'a']
['h' 'm' 'o' 'u' 'd' 'm' 'm' 'a']
['o' 'u' 'd' 'm' 'm' 'a']
['d' 'm' 'm' 'a']
['m' 'a']
[]


35. Accept string from the user and display only those characters which are present at an even
index

In [13]:
st=input('Enter a string: ')
print(st[0::2])

Enter a string: mahmoud
mhod


36. Remove duplicate from a string and print the results.

In [20]:
st='mahmoud'
b=list(st)
b=list(dict.fromkeys(b)) 

st=str(b)
print(st)

['m', 'a', 'h', 'o', 'u', 'd']


37. Remove duplicate from a string and print how much time each character appeared.

In [50]:
sts=input('Enter a string ')
st=sts
st1=''
s=0
d=0
b =list()
m=0
b1=list(st)
b1=list(dict.fromkeys(b1)) 

st2=str(b1)
print(st2)
for n in st:
    c=st.count(n)
    b.append(int(str(c)))
    if(c>1):
        st=st.replace(n,'')
b = [x for x in b if x != 0]
print(b)

Enter a string mghmgg
['m', 'g', 'h']
[2, 3, 1]


38. Find all unique character, numbers, ...etc. in a string and print them.

In [59]:
sts=input('Enter a string ')
st=sts
b=[]
for n in st:
    c=st.count(n)
    if(c==1):
        b.append(n)
        st=st.replace(n,'')
    else:
        pass
        
print(b)
        

Enter a string dsnv;sd63
['n', 'v', ';', '6', '3']


39. Write a Python program to accept a filename from the user as a string and print the extension
of that file.
Sample filename : abc.java
Output : java

In [63]:
st=input('Enter a file name with its extension:')
st1=st.split('.')
st1=repr(st1[-1])
print(st1)

Enter a file name with its extension:mahmoud.java
'java'


40. Write a Python program to get a new string from a given string where ”Is” has been added to
the front. If the given string already begins with ”Is” then return the string unchanged.

In [67]:
st=input('Enter a string: ')
st1='Is'
st2=st[:2]
if st2==st1 :
    print(' the string unchanged')
else:
    st3=st1+st
    print(st3)


Enter a string: Is mahmoud
 the string unchanged


41. Write a Python program to get a string which is n (non-negative integer) copies of a given
string

In [69]:
st=input('Enter a string: ')
while True:
    try:
        n=int(input('Enter a non-negative integer only:'))
        if n or n>=0:
            break
    except:
        print('Enter a valid non-negative integer ')
print(st*n)

Enter a string: python . 
Enter a non-negative integer only:5
python . python . python . python . python . 


42. Write a Python program to get the n (non-negative integer) copies of the first 2 characters of
a given string. Return the n copies of the whole string if the length is less than 2.

In [74]:
st=input('Enter a string: ')
while True:
    try:
        n=int(input('Enter a non-negative integer only:'))
        if n or n>=0:
            break
    except:
        print('Enter a valid non-negative integer ')
b = len(st)
if b<2:
    print(st*n)
else:
    print(st[:2]*n)

Enter a string: msa
Enter a non-negative integer only:5
msmsmsmsms


43. Write a Python program to print out the set of all the characters from St1 which are not present
in St2

In [85]:
# st=input('Enter a string st1: ')
# st2=input('Enter a string st2: ')
# m=len(st)
# lis=''
# b=len(st2)
# n=1
# a=1
# for n in range(m):
#     if st2[(n-1)] in st1:
#         pass
#     else:
#         lis=lis+st2[(n-1)]
#         #add(lis,st2[n])
# for a in range(b):
#     if st1[(a-1)] in st2:
#         pass
#     else:
#         lis=lis+st2[(a-1)]
#         #add(lis,st1[n])
# print(lis)
    

                
    

44. Write a Python program to display your given details like name, age, address in three different
lines.

In [87]:
name='mahmoud alqannas'
age=25
address = 'Amman, Jordan'

print("Name: {}\nage: {}\nAddress: {}".format(name, age, address))


Name: mahmoud alqannas
age: 25
Address: Amman, Jordan


45. Write a Python program to sort three integers without using conditional statements, loops or
strings.

In [96]:
lis=[5,94,78]
lis.sort()
print(lis)


[5, 78, 94]


46. Write a Python program to get the summation of the ASCII values of aaa characters in a string.

47. Write a Python program to convert a byte string to a list of integers.

In [99]:
st1= b'mahmoud'   
print(list(st1))


[109, 97, 104, 109, 111, 117, 100]


48. Write a Python program to get the identity of an object.

In [101]:
st='nah'
id(st)


2875463995496

49. Write a Python program to get the system time.

In [103]:
import time
t= time.localtime()
time = time.strftime("%H:%M:%S", t)
print(time)

15:25:35


50. Write a Python program to find the ASCII string of a given number.

In [109]:
b=97
print(chr(b))

a


51. Write a Python program to prove that two string variables of same value point same memory
location.

In [110]:
st1='mahmoud'
st2='mahmoud'
if id(st1)==id(st2):
    print('st1 and st2 are in the same memory location')

st1 and st2 are in the same memory location


    52. Write a Python function that takes a positive integer and returns the sum of the cube of all
the positive integers smaller than the specified number.

In [114]:
while True:
    try:
        n=int(input('Enter a non-negative integer only:'))
        if n or n>=0:
            break
    except:
        print('Enter a valid postive integer ')
sum1=0
for i in range(n):
    sum1=sum1+i**3
print(sum1)
    

Enter a non-negative integer only:4
36


53. Write a Python program to add two positive integers without using the ’+’ operator.
Note: Use bit wise operations to add two numbers

In [126]:
a=5
b=9
while b != 0:
        data = a & b
        a = a ^ b
        b = data << 1
print(a)

14


54. Write a Python program to get all strobogrammatic numbers that are of length n.
A strobogrammatic number is a number whose numeral is rotationally symmetric, so that it
appears the same when rotated 180 degrees. In other words, the numeral looks the same rightside
up and upside down (e.g., 69, 96, 1001).
For example,
Given n = 2, return [”11”, ”69”, ”88”, ”96”]. Given n = 3, return [’818’, ’111’, ’916’, ’619’,
’808’, ’101’, ’906’, ’609’, ’888’, ’181’, ’986’, ’689’]

In [134]:
def gen_strobogrammatic(n):
    """
    :type n: int
    :rtype: List[str]
    """
    result = helper(n, n)
    return result


def helper(n, length):
    if n == 0:
        return [""]
    if n == 1:
        return ["1", "0", "8"]
    middles = helper(n-2, length)
    result = []
    for middle in middles:
        if n != length:
            result.append("0" + middle + "0")
        result.append("8" + middle + "8")
        result.append("1" + middle + "1")
        result.append("9" + middle + "6")
        result.append("6" + middle + "9")
    return result

print("n = 2: \n",gen_strobogrammatic(2))
print("n = 3: \n",gen_strobogrammatic(3))
print("n = 4: \n",gen_strobogrammatic(4))

n = 2: 
 ['88', '11', '96', '69']
n = 3: 
 ['818', '111', '916', '619', '808', '101', '906', '609', '888', '181', '986', '689']
n = 4: 
 ['8008', '1001', '9006', '6009', '8888', '1881', '9886', '6889', '8118', '1111', '9116', '6119', '8968', '1961', '9966', '6969', '8698', '1691', '9696', '6699']


55. Write a Python program to find the median among n given numbers.

In [144]:
import statistics
lis=[5,9,8,7,6,41,85,265,74]
print(statistics.median(lis))


9


56. Write a Python program to find all the divisors of a given number.

In [150]:
n=50
s=0
b=[]

for i in range(1,50):
    s=50%i
    if s==0:
        b.append(i)
print(b)

[1, 2, 5, 10, 25]


57. Write a Python program to find the number of divisors of a given integer is even or odd.

In [151]:
n=50
s=0
b=[]

for i in range(1,50):
    s=50%i
    if s==0:
        b.append(i)
c=len(b)
print(b)
if c%2==0:
    print('the number of divisors is even')
else:
    print('the number of divisors is odd')
        

[1, 2, 5, 10, 25]
the number of divisors is odd


58. Write a Python program to create a sequence where the first four members of the sequence are
equal to one, and each successive term of the sequence is equal to the sum of the four previous
ones. Find the Nth member of the sequence.

In [5]:
while True:
    try:
        x=int(input('Enter a Nth member of the sequence you want: '))
        if x or x>0:
            break
    except:
            print('Enter avalid number:')
b=[]*(4**x)
c=len(b)
v=4**x
for n in range(0,v,4):
    
    for s in range(1,5):
        b.append()
        
print(b)

Enter a Nth member of the sequence you want: 4


IndexError: list index out of range

59. Write a Python program that accept a positive number and subtract from this number the sum
of its digits and so on. Continues this operation until the number is positive

In [10]:
while True:
    try:
        x=int(input('Enter a number: '))
        if x or x == 0:
            break
    except:
        print('Enter a valid number:')
dev=0
temp=x
d=0
s=0
while (x>0):
    dig=x%10
    dev=dev*10+dig
    s=s+dev%10
    d+=1
   # print(dev)
    x//=10
print('sum of its digits : ',s)
while temp>0:
    print(temp)
    temp=temp-s


Enter a number: 100
sum of its digits :  1
100
99
98
97
96
95
94
93
92
91
90
89
88
87
86
85
84
83
82
81
80
79
78
77
76
75
74
73
72
71
70
69
68
67
66
65
64
63
62
61
60
59
58
57
56
55
54
53
52
51
50
49
48
47
46
45
44
43
42
41
40
39
38
37
36
35
34
33
32
31
30
29
28
27
26
25
24
23
22
21
20
19
18
17
16
15
14
13
12
11
10
9
8
7
6
5
4
3
2
1


60. Write a Python program to find the digits which are absent in a given mobile number.

In [19]:
n=[5,0,9,8,7,4,6]
all_nums = set([0,1,2,3,4,5,6,7,8,9])
n = set([int(i) for i in n])
n = n.symmetric_difference(all_nums)
n = sorted(n)
print(n)    

[1, 2, 3]


Write a Python program to check whether three given lengths (integers) of three sides form a
right triangle. Print ”Yes” if the given sides form a right triangle otherwise print ”No”.
Input: Integers separated by a single space.
1 = length of the side = 1,000
Input three integers(sides of a triangle)
8 6 7
No